modules/services/ihaskell.nix

```diff
@@ -0,0 +1,78 @@
+# Adapted from Nixpkgs.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.ihaskell;
+  ihaskellSrc = pkgs.fetchFromGitHub {
+    owner = "gibiansky";
+    repo = "IHaskell";
+    rev = "c2cb8e6789c3f2485ba9e1e0436177b700edb227";
+    sha256 = "0h5hpdfa3daabf1ijl7y5qnpy60yrf7h0ws2n3v5przfmgr8phv5";
+  };
+  ihaskellResult = (import (ihaskellSrc + /release.nix) {
+    compiler = "ghc864";
+    packages = self: cfg.extraPackages self;
```

Comment:
There are also `pythonPackages` and (badly named) `systemPackages` attributes. What do you think about making those configurable?

Reply:
Also, you can replace this with just `cfg.extraPackages` and it should work the same :smile:.

Reply:
Any reason someone would want to customize `pythonPackages` or `systemPackages` when using ihaskell?

Reply:
One example is using Z3 from IHaskell.